### PR TITLE
Soft Shadows and Fog [Preview]

### PR DESCRIPTION
**Pull request by gabrielmtzcarrillo**
## Fixes
- Icon not added while compiling for windows and audio config being copied to another directory.
- Zero intensity lights not being treated as default light (which is vec4(1.0) in TR1?).
- Crash related to max animation frame on TRLE Levels.

## Changes
- Softer lightning.
- Brighter colours for TR1.

## Added
- Distance Fog.
- Missing TRLA Braids.

### TR1
screen_20170926_144820 png
screen_20170926_145246 png
screen_20170926_145402 png
screen_20170926_145427 png
### TRLE - Coyote Creek
screen_20170927_110509 png
screen_20170927_111235 png

### CODE REVIEW

File: shaders/entity.fsh [hunk 1]
```diff
@@ -16,29 +16,47 @@ varying vec2 varying_texCoord;
...
 {
     // Find color from lights
-    vec4 lightColor = light_ambient;
+	vec4 lightAmbient;
+    vec4 lightColor = vec4(0.0);
+	vec3 lightDirection = vec3(0.0,1.0,0.0);
     vec3 normal = normalize(varying_normal);
     
 #if NUMBER_OF_LIGHTS > 0
+	vec4 tmp;
```
Comment (vvs-) [inline shaders/entity.fsh]: This doesn't compile: 0:59(7): error: `tmp' redeclared

File: src/config-opentomb.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+//
+
+#ifndef config_opentomb_h_in_h
+#define config_opentomb_h_in_h
+
+#define HAVE_ALC_H 1
+#define HAVE_EFX_H 1
+#define HAVE_EFX_PRESETS_H 1
+
+#endif /* config_opentomb_h_in_h */
```
Comment (vvs-) [inline src/config-opentomb.h]: This file is autogenerated and shouldn't be in repository.

**Comment by gabrielmtzcarrillo** [comment]
Forgot to copy my working shaders

**Comment by vvs-** [comment]
Works perfectly. Looks like you finally managed to fix those lighting bugs. And distance fog is nice indeed. 👍

**Comment by gabrielmtzcarrillo** [comment]
The fog part was kind of last minute while i was looking for a water-y fog thingy effect.

**Comment by TeslaRus** [comment]
thanks for fixes and PR =)
lgtm, but before merge, please, replace (convert) tabs to spaces (1 tab == 4 spaces)

**Comment by gabrielmtzcarrillo** [comment]
You're welcome!, it's nice to see projects like this.

**Comment by vvs-** [comment]
@gabrielmtzcarrillo Please, don't forget to tell us when you are ready to merge it.

**Comment by Gh0stBlade** [comment]
@gabrielmtzcarrillo Just a heads up. TR5 lights are seriously broken. The crux of the issue is more-so the way the data is read into the engine (it's incorrect).

File: src/gui/gui.cpp [hunk 3]
```diff
@@ -1712,6 +1712,15 @@ void gui_ItemNotifier::Draw()
         base_item_p item = World_GetBaseItemByID(mItem);
         if(item)
         {
+            gl_text_line_p inventory_name = GLText_OutTextXY(mCurrPosX, 45.0f, "%s", item->name);
```
Comment (TeslaRus) [inline src/gui/gui.cpp]: upd: this time UI and core (font, text, console) modules is under heavy refactoring (it is better to reduce changes here before gui_refactoring branch merging).
inventory manager has it's own gl_text_line_p items

File: scripts/level/tr3/CHUNNEL.lua [hunk 4]
```diff
@@ -4,7 +4,7 @@
 print("Level script loaded (CHUNNEL.lua)");
```
Comment (TeslaRus) [inline scripts/level/tr3/CHUNNEL.lua]: nice! I think it is better to use 2 - 5 little PR instead one big - much more faster review and less merge conflicts, more faster master update.

**Comment by gabrielmtzcarrillo** [comment]
@Gh0stBlade , I've been fiddling with lights and I think that a global illumination shader is nedded instead of different shaders, especially for dynamic lightning (flares and torchs).

@TeslaRus, I'll revert f50f670e4abc95ba76b7ee5c0be0f9b75ce45b87

**Comment by TeslaRus** [comment]
@gabrielmtzcarrillo thanks =) , hope today I will update remote feature branch (to preview first changes)

**Comment by vvs-** [comment]
@gabrielmtzcarrillo OpenLara has performance issues exactly because of its ubershader structure. It's bad for low end GPUs, like mobile devices or integrated desktop graphics.

**Comment by gabrielmtzcarrillo** [comment]
@vvs- , as far I've seen, I think the problem relies on using Ambient Occlusion (AO) offloaded to CPU instead of GPU via shaders.

**Comment by vvs-** [comment]
@gabrielmtzcarrillo You might know better. I've tried to profile it but can't progress too far without deep OpenGL knowledge.

**Comment by gabrielmtzcarrillo** [comment]
@vvs- , honestly, this is the first time that I actually work with OpenG; I'm more used to C# and XNA, but their core mechanics are similar.

**Comment by TeslaRus** [comment]
May be merge this PR in current state? Next updates will be in next PR, because it tends that current PR is permanent...

**Comment by vvs-** [comment]
Shaders are somewhat thrashed:
screen_20171009_154725

**Comment by TeslaRus** [comment]
ok, thanks - entity shader is wrong (or missed changes)

**Comment by TeslaRus** [comment]
shaders was fixed in master

**Pull request merged**